src/openforms/submissions/api/validation.py

```diff
@@ -78,6 +85,27 @@ def validate_submission_completion(
     completion = InvalidCompletion(
         submission=submission,
         incomplete_steps=incomplete_steps,
+        invalid_prefilled_fields=invalid_prefilled_fields,
         request=request,
     )
     return completion.validate()
+
+
+def get_invalid_prefilled_fields(
+    submission_step: "SubmissionStep", submission: "Submission"
+) -> List[str]:
+    invalid_prefilled_fields = []
+
+    for component in submission_step.form_step.iter_components():
+        if "prefill" not in component or component["prefill"]["plugin"] == "":
+            continue
+
```

Comment:
Should this check also not only apply if the component is marked as disabled/readonly? If you expose the disabled/readonly controls to form-designers, this would lead to unexpected results otherwise where they may allow prefilled fields to be altered?

Reply:
Good point